dev/src/Split/GitHub.php

```diff
@@ -0,0 +1,142 @@
+ *
+ * @internal
+ */
+class GitHub
+{
+    const GITHUB_RELEASES_ENDPOINT = 'https://api.github.com/repos/%s/releases/tags/%s';
+    const GITHUB_RELEASE_CREATE_ENDPOINT = 'https://api.github.com/repos/%s/releases';
+
+    /**
+     * @var RunShell
+     */
+    private $shell;
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var string
+     */
+    private $token;
+
+    public function __construct(RunShell $shell, Client $client, $token)
+    {
+        $this->shell = $shell;
+        $this->client = $client;
+        $this->token = $token;
+    }
+
+    /**
+     * Check if a given tag exists on a given GitHub repository.
+     *
+     * @param string $target The GitHub organization and repository ID separated
+     *        by a forward slash, i.e. `organization/repository'.
+     * @param string $tagName The tag to check.
+     * @return bool
+     */
+    public function doesTagExist($target, $tagName)
+    {
+        $res = $this->client->get(sprintf(
+            self::GITHUB_RELEASES_ENDPOINT,
+            $this->cleanTarget($target), $tagName
+        ), [
+            'http_errors' => false,
+            'auth' => [null, $this->token]
+        ]);
+
+        return ($res->getStatusCode() === 200);
+    }
+
+    /**
+     * Create a tag on the given GitHub repository and sends a request to verify.
+     *
+     * @param string $target The GitHub organization and repository ID separated
+     *        by a forward slash, i.e. `organization/repository'.
+     * @param string $tagName The name of the tag to create.
+     * @param string $display The tag display name.
+     * @param string $notes The tag release notes.
+     * @return bool True if successful, false if failed.
+     */
+    public function createRelease($target, $tagName, $display, $notes)
+    {
+        $requestBody = [
+            'tag_name' => $tagName,
+            'name' => $display,
+            'body' => $notes
+        ];
+
+        $res = $this->client->post(sprintf(
+            self::GITHUB_RELEASE_CREATE_ENDPOINT,
+            $this->cleanTarget($target)
+        ), [
+            'http_errors' => false,
+            'json' => $requestBody,
+            'auth' => [null, $this->token]
+        ]);
+
+        return $this->doesTagExist($target, $tagName);
+    }
+
+    /**
+     * Push a given commit ref to GitHub.
+     *
+     * @param string $target The GitHub organization and repository ID separated
+     *        by a forward slash, i.e. `organization/repository'.
+     * @param string $ref The commit reference.
+     * @param string $targetBranch The remote branch to push to. **Defaults to**
+     *        `master`.
+     * @param bool $force If true, will use `--force` flag. **Defaults to**
+     *        `true`.
+     * @return array A list containing [(bool) $success, (string) $output].
+     */
+    public function push($target, $ref, $targetBranch = 'master', $force = true)
+    {
+        $cmd = 'git push -q ' .
```

Comment:
Since it's easy to mess up the space formatting in things like this, I might make an array and join them with a space, especially when I'm building up a command with optional args. We've only got the one for `--force`, but I like this:

```php
$parts - ['git push', '-q', 'https://%s@github.com/%s', '%s:%s'];

if ($force) {
  $parts[] = ['--force'];
}

$cmd = join($parts, ' ');
```

Just a style thing because I've got tired of fixing formatting problems, but this one looks right